include/range/v3/utility/array.hpp

```diff
@@ -0,0 +1,304 @@
+            const value_type* data() const noexcept
+            {
+                return elems_;
+            }
+        };
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator==(const array<T, N>& x, const array<T, N>& y)
+        {
+            return ranges::equal(x.elems_, x.elems_ + N, y.elems_);
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator!=(const array<T, N>& x, const array<T, N>& y)
+        {
+            return !(x == y);
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator<(const array<T, N>& x, const array<T, N>& y)
+        {
+            return ranges::lexicographical_compare(x.elems_, x.elems_ + N, y.elems_, y.elems_ + N);
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator>(const array<T, N>& x, const array<T, N>& y)
+        {
+            return y < x;
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator<=(const array<T, N>& x, const array<T, N>& y)
+        {
+            return !(y < x);
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator>=(const array<T, N>& x, const array<T, N>& y)
+        {
+            return !(x < y);
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        auto swap(const array<T, N>& x, const array<T, N>& y)
+                noexcept(is_nothrow_swappable<T, T>::value)
+                -> typename std::enable_if<is_swappable<T, T>::value, void>::type
+        {
+            x.swap(y);
+        }
+
+
+        template <size_t I, class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        T& get(array<T, N>& a) noexcept
+        {
+            static_assert(I < N, "Index out of bounds in ranges::get<> (ranges::array)");
+            return a.elems_[I];
+        }
+
+        template <size_t I, class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        const T& get(const array<T, N>& a) noexcept
+        {
+            static_assert(I < N, "Index out of bounds in ranges::get<> (const ranges::array)");
+            return a.elems_[I];
+        }
+
+        template <size_t I, class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        T&& get(array<T, N>&& a) noexcept
+        {
+            static_assert(I < N, "Index out of bounds in ranges::get<> (ranges::array &&)");
+            return std::move(a.elems_[I]);
+        }
+
+    }  // namespace v3
+}  // namespace ranges
+
+namespace std
+{
+
+template <class T, size_t N>
+class tuple_size<ranges::array<T, N>>
+    : public integral_constant<size_t, N> {};
+
+template <size_t I, class T, size_t N>
+class tuple_element<I, ranges::array<T, N> >
+{
+ public:
+    using type = T;
+};
+
+}  // namespace std
+
+#endif // RANGES_V3_UTILITY_ARRAY_HPP
```

Comment:
I guess you mean the duplication of `std::array` in general (correct me if I'm wrong). 

The duplication is not necessary, but it simplifies testing so much that it is worth having. As stated in the pull request, I'm happy to move this to the test directory. The issue it solves is that besides C-arrays, and initializer lists, I need to test against a constexpr friendly user defined sequence. This array class is way simpler than `view::ints` to get right, and adds almost no dependencies. At the beginning I was using `view::ints` to test everything, but that pulls more of range-v3 into each test and requires a lot of `view`-related functionality to be constexpr. I didn't want to submit all those changes just yet. I've done it once, and I prefer to tackle this step by step, in the following order: core, algorithms, views, and finally actions.

So I guess I agree, it sucks that `std::array` is broken, otherwise I would use it. But this is better than the other alternatives I've tried, so I would like to keep it.